src/ray/pubsub/publisher.cc

```diff
@@ -207,24 +249,31 @@ void Publisher::Publish(const ObjectID &object_id) {
     auto it = subscribers_.find(subscriber_id);
     RAY_CHECK(it != subscribers_.end());
     auto &subscriber = it->second;
-    subscriber->QueueMessage(object_id);
+    subscriber->QueueMessage(std::move(pub_message));
```

Comment:
If the size of `maybe_subscribers` is 2, `pub_message` will be moved twice, which may lead segement fault.

Reply:
Good catch. Will fix it and add tests

Reply:
@rkooo567 This still needs to be fixed, right?

Reply:
Yep! Haven't addressed @jovany-wang's last comments.